src/bin/odata-compiler.ts

```diff
@@ -0,0 +1,125 @@
+process.env.PINEJS_CACHE_FILE =
+	process.env.PINEJS_CACHE_FILE || __dirname + '/.pinejs-cache.json';
+
+import * as _abstractSql from '../sbvr-api/abstract-sql';
+import * as _sbvrUtils from '../sbvr-api/sbvr-utils';
+import * as _uriParser from '../sbvr-api/uri-parser';
+
+import * as program from 'commander';
+import * as fs from 'fs';
+import '../server-glue/sbvr-loader';
+
+// tslint:disable:no-var-requires
+const { version } = JSON.parse(
+	fs.readFileSync(require.resolve('../../package.json'), 'utf8'),
+);
+
+const getSE = (inputFile: string) => fs.readFileSync(inputFile, 'utf8');
+
+const generateAbstractSqlModel = (sbvrFile: string) => {
+	const {
+		generateLfModel,
+		generateAbstractSqlModel,
+	} = require('../sbvr-api/sbvr-utils') as typeof _sbvrUtils;
```

Comment:
i guess that we use `require()` and "type cast" so that we have sync lazy loading?
(Correct me if I'm wrong but I couldn't guess any other reason to not use `import from` / `await import()`).

Reply:
Yeah, it's to keep things sync so errors are reported nicely, commander doesn't support promises and I don't feel a need to learn/rewrite in another command line parser (this is ported in/converted to typescript from https://github.com/balena-io-modules/odata-compiler just to make it easier to use with the right deps within a pinejs project)